server/app/routes/order/index.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+var router = require('express').Router()
+var db = require('../../../db')
+var Order = db.model('order')
+module.exports = router
+
+//Get all orders
+router.get('/', function(req, res, next) {
+  Order.findAll({where: req.query})
+  .then(function(orders) {
+    res.send(orders)
+  })
+  .catch(next)
+})
+
+//Get one order by id
+router.get('/:orderId', function(req, res, next) {
```

Comment:
since all the routing for /:orderId make the same query of Order.findById, and require the same error throwing,

maybe we can set up one middleware that handles all requests to /:orderId.

```
router.param('orderId', function (req, res, next, orderId) {
    Order.findById(orderId)
    .then(function (order) {
        if (order) {
            req.order = order;
            next();
            return null;
        } else {
            var error = new Error('some message');
            error.status = 404;
            throw error;
        }
    })
    .catch(next);
});
```

This way could be easier so if we ever make any changes, we would only have to change them in one place.

Also, the error status of 404 is appropriate for get, put, and delete, since that orderId was 'Not Found.'

Reference (from juke solution) 

```
router.get('/', function (req, res, next) {
  Playlist.findAll({ where: req.query })
  .then(playlists => res.json(playlists))
  .catch(next);
});

router.post('/', function (req, res, next) {
  Playlist.create(req.body)
  .then(playlist => res.status(201).json(playlist))
  .catch(next);
});

router.param('playlistId', function (req, res, next, id) {
  Playlist.scope('populated').findById(id)
  .then(playlist => {
    if(!playlist) throw new Error('not found!');
    req.playlist = playlist;
    next();
    return null; // silences bluebird warning about promises inside of next
  })
  .catch(next);
});

router.get('/:playlistId', function (req, res) {
  res.json(req.playlist);
});

router.put('/:playlistId', function (req, res, next) {
  req.playlist.update(req.body)
  .then(playlist => res.status(200).json(playlist))
  .catch(next);
});
```